external-plugins/referee/server/server.go

```diff
@@ -117,17 +117,59 @@ func (s *Server) handleEvent(eventType, eventGUID string, payload []byte) error
 		if err := json.Unmarshal(payload, &ic); err != nil {
 			return err
 		}
-		go func() {
+		go func(ic github.IssueCommentEvent) {
 			if err := s.handlePullRequestComment(ic); err != nil {
 				s.Log.WithError(err).WithFields(l.Data).Errorf("%s failed.", PluginName)
 			}
-		}()
+		}(ic)
+	//https://developer.github.com/webhooks/event-payloads/#pull_request
+	case "pull_request":
+		var pr github.PullRequestEvent
+		if err := json.Unmarshal(payload, &pr); err != nil {
+			return err
+		}
+		go func(pr github.PullRequestEvent) {
+			if err := s.handlePREvent(pr); err != nil {
+				s.Log.WithError(err).WithFields(l.Data).Errorf("%s failed.", PluginName)
+			}
+		}(pr)
 	default:
 		s.Log.WithFields(l.Data).Debugf("skipping event of type %q", eventType)
 	}
 	return nil
 }
 
+func (s *Server) handlePREvent(pr github.PullRequestEvent) error {
+	org := pr.Repo.Owner.Login
+	repo := pr.Repo.Name
+	num := pr.Number
+	action := pr.Action
+
+	pullRequestURL := fmt.Sprintf("https://github.com/%s/%s/pull/%d", org, repo, num)
+	log := s.Log.WithField("pull_request_url", pullRequestURL)
+
+	switch action {
+	case github.PullRequestActionOpened:
```

Comment:
Is this case required?

Reply:
Yes, to initiate a new PR with zero retests, otherwise the new one will only show up after the first retest.